simple_producer.go

```diff
@@ -14,15 +14,17 @@ type spExpect struct {
 	result chan error
 }
 
-// NewSimpleProducer creates a new SimpleProducer using the given client, topic and partitioner. If the
-// partitioner is nil, messages are partitioned by the hash of the key
-// (or randomly if there is no key).
-func NewSimpleProducer(client *Client, topic string, partitioner PartitionerConstructor) (*SimpleProducer, error) {
+// NewSimpleProducerWithConfig is the same as NewSimpleProducer, except it
+// includes the option to set certain configuration parameters on the
+// underlying producer
+func NewSimpleProducerWithConfig(client *Client, topic string, partitioner PartitionerConstructor, config *ProducerConfig) (*SimpleProducer, error) {
```

Comment:
I'm not a huge fan of the `WithConfig` suffix, but I can't come up with anything better off the top of my head.